internal/pkg/service/buffer/worker/service/cleanup.go

```diff
@@ -0,0 +1,90 @@
+package service
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/log"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/worker/distribution"
+)
+
+type cleanup struct {
+	*Service
+	logger   log.Logger
+	assigner *distribution.Assigner
+
+	lock *sync.RWMutex
+}
+
+func (s *Service) cleanup(ctx context.Context, wg *sync.WaitGroup) <-chan error {
+	return s.dist.StartWork(ctx, wg, s.logger, func(ctx context.Context, assigner *distribution.Assigner) (initDone <-chan error) {
+		return startCleanup(ctx, wg, s, assigner)
+	})
+}
+
+func startCleanup(ctx context.Context, wg *sync.WaitGroup, s *Service, assigner *distribution.Assigner) <-chan error {
+	c := &cleanup{
+		Service:  s,
+		logger:   s.logger.AddPrefix("[cleanup]"),
+		assigner: assigner,
+		lock:     &sync.RWMutex{},
+	}
+
+	// Start ticker
+	startTime := c.clock.Now()
+	c.startTicker(ctx, wg)
+	c.logger.Infof(`initialized | %s`, c.clock.Since(startTime))
+	initDone := make(chan error, 1)
+	defer close(initDone)
+	return initDone
+}
+
+// startTicker to check conditions periodically.
+func (c *cleanup) startTicker(ctx context.Context, wg *sync.WaitGroup) {
+	wg.Add(1)
```

Comment:
Toto by nutne nemuselo byt v samostatnej funcii, kedze to je jednine co sa robi.

Osobne si myslim, ze funkcie `startCleanup` a `startTicker` nepotrebujes, cele to krasne moze vojst do `(s *Service) cleanup`, na inych miestach sa robi vacsinou viacero veci, tak je to tam viac podelene. Ale to len poznamka bokom.